Assignment 3.Rmd

```diff
@@ -127,27 +135,168 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
+locfit <- kmeans(DF2, 4) 
+DF2.1 <- data.frame(DF2, locfit$cluster)
+
+#Easier column names
+names(DF2.1) <- c("long", "lat", "cluster") 
+
+#Cluster colors are more distinct as factors
+DF2.1$cluster <- as.factor(DF2.1$cluster)
+
+#Plot region clusters
+ggplot(DF2.1, 
+       aes(x = DF2.1$long,
+           y = DF2.1$lat,
+           color = DF2.1$cluster,
+           shape = DF2.1$cluster))+
+  geom_point(size = 4)+
+  scale_shape_discrete(solid=F)+
+  theme(plot.title = element_text(hjust = 0.5,
+                                  family = "Helvetica",
+                                  face="bold"))+
+  labs(x = "Longitude",
+       y = "Latitude",
+       color = "Region Cluster",
+       shape = "Region Cluster",
+       title = "Region Clusters by Location")
+```
+
+#Activity CLusters
+
+Create Activity Data Frame
+```{r}
+#Omit City and Country columns
+DF3 <- DF1.1[,-1:-2]
+
+#Convert time values to numeric
+columns <- names(DF3)
+DF3[, columns] <- lapply(columns, function(x) as.numeric(DF3[[x]]))
+```
+
+Form Activity Clusters
+```{r}
+#Create k means clusters by activity
+actfit <- kmeans(DF3, 4)
+
+#Attach activity clusters to data frame DF3.1
+DF3.1 <- data.frame(DF3, actfit$cluster)
+
+#Assign easer column names
+colnames(DF3.1) <- c(c(1:7),"cluster")
+
+#Convert data to long format
+DF3.2 <- gather(DF3.1,'act','time',1:7)
+
+#Average times by activity and activity cluster
+DF3.3 <- DF3.2 %>% group_by(act,cluster)
+DF3.3 <- summarise(DF3.3,avg=mean(time))
+
+#Convert data to desirable class
+DF3.3$act <- as.numeric(DF3.3$act)
+DF3.3$cluster <- as.factor(DF3.3$cluster)
+
+#Plot activity clusters
+ggplot(DF3.3,
+       aes(act,avg,color=cluster))+
+  geom_line()+
+  theme(plot.title = element_text(hjust = 0.5, 
+                                  family = "Helvetica", 
+                                  face="bold"))+
+  labs(x = "Activity:1=Code, 2=Write, 3=Acd Read/Vids, 4=Vids4Fun, 5=Cooking, 6=SNS, 7=Going Out",
+       y = "Average Time (hr)",
+       title = "Average Activity Time per Cluster",
+       color = "Cluster\n")
+```
+
 ##Part III
 
 Create a visualization that shows the overlap between the two groups of clusters you created in part III.
-
+```{r}
+#Combines DF2.1 and DF3.1 into DF4
+DF4 <- data.frame(DF2.1,DF3.1)
+
+#Change column names
+colnames(DF4)[3] <-"loc_cluster"
+colnames(DF4)[11] <-"act_cluster"
+
+#Convert to factor
+DF4$act_cluster <- as.factor(DF4$act_cluster)
+DF4$loc_cluster <- as.factor(DF4$loc_cluster)
+
+#Plot Region and Activity Clusters by Location
+ggplot(DF4,
+       aes(x = long,
+           y = lat,
+           color = DF4$loc_cluster,
+           shape = DF4$act_cluster))+
+  geom_point(size = 4)+
+  scale_shape_discrete(solid=F)+
+  theme(plot.title = element_text(hjust = 0.5,
+                                  family = "Helvetica",
+                                  face="bold"))+
+  labs(x = "Longitude",
+       y = "Latitude",
+       title = "Region and Activity Clusters by Location",
```

Comment:
A beautiful plot! Well done! Does it give a good reflection of which groups of clusters overlap though?